### PR TITLE
Update ChannelsProvider to use async queries

### PR DESCRIPTION
**Pull request by m-rosario**
- Update the remaining queries inside ChannelsProvider to return Future instead of Widget
- Updated the query for the channels last message to use instead Channel.latestMessage
- Some miscellaneous house keeping in related classes

Note that I tried to also implement an improved version of the query for unseen messages, but it seems that the corresponding backend query is not working as expected. This will be revised later.

### CODE REVIEW

File: lib/widgets/screens/channel_messages/channel_messages.dart [hunk 1]
```diff
@@ -117,17 +136,16 @@ class _ChannelChatState extends State<ChannelChat> {
   final ScrollController listScrollController = ScrollController();
   final Duration _animationDuration = const Duration(milliseconds: 250);
   ChatModel? _chatModel;
+  MessagesProvider? _messagesProvider;
```
Comment (jupblb) [inline lib/widgets/screens/channel_messages/channel_messages.dart:139]: You may mark it as `late final` and skip the `?`.

**Review by jupblb — DISMISSED** [review]

File: lib/widgets/molecules/invitation_section.dart [hunk 2]
```diff
@@ -21,43 +23,65 @@ class InvitationSection extends StatefulWidget {
...
+  void initState() {
+    super.initState();
+    _channelsProvider = Provider.of<ChannelsProvider>(context, listen: false);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _invitationInbox = _channelsProvider?.getInboxInvitations();
+    l10n = AppLocalizations.of(context);
```
Comment (LucasMagnum) [inline lib/widgets/molecules/invitation_section.dart:40]: Is the context available here?

File: lib/widgets/screens/channel_messages/channel_messages.dart (hunk 1 above)
Comment (m-rosario) [reply lib/widgets/screens/channel_messages/channel_messages.dart]: You're super right! Replaced this and other declarations to use lazy initializer.

File: lib/widgets/molecules/invitation_section.dart (hunk 2 above)
Comment (m-rosario) [reply lib/widgets/molecules/invitation_section.dart:40]: Yes, the BuildContext is an instance field of the State class. It is available right after initialization.

**Review by LucasMagnum — APPROVED** [review]

**Pull request merged**